tests/python_functional/src/syslog_ng_config/statements/sources/network_source.py

```diff
@@ -26,14 +26,20 @@
 
 class NetworkSource(SourceDriver):
     def __init__(self, **options):
-        self.io = self.__create_io(options)
-
         self.driver_name = "network"
-        super(NetworkSource, self).__init__(options=options)
+        self.io = self.__create_io(options)
+        super(NetworkSource, self).__init__(logger_name=__name__, options=options)
 
     def write_log(self, formatted_content):
         self.io.write(formatted_content)
 
+        super(NetworkSource, self).write_log(
+            formatted_content,
+            transport=self.io.get_transport(),
```

Comment:
I am not really understand the borders what you made between the Config vs NetworkIO layer.
From my point of view it is fine to grab transport value, from here. So I would write this:
`transport = self.options['transport']` instead of `transport=self.io.get_transport(),`
And also in case of ip and port. Could you please explain your borders (in this actual case)?

Reply:
finally I understand.